examples/acoustics/acoustic.py

```diff
@@ -0,0 +1,179 @@
+    plt.savefig(f'{name}_narrow-band_spectra_{key}.png', bbox_inches='tight')
+    plt.close()
+
+def getSpectrum(aSound,rho,acsv,oName):
+    # reference sound pressure in air
+    p_ref = 2e-5
+
+    # reads acoustic csv file
+    tt,stp,pdata = readAcousticCSVFile(acsv)
+    # filters zeros from acoustic data
+    acoustic_data = filterZeroAcousticData(tt,stp,pdata)
+    # number of observers based on acoustic data
+    numObservers = len(acoustic_data)-2
+
+    # output file basename
+    outFileName = os.path.splitext(os.path.basename(oName))[0]
+    # output oaspl file in dat format
+    oasplFile = os.path.join(os.curdir,outFileName+'_OASPL'+'.dat')
+    print(f"variables=i,{'oaspl_'+outFileName}",file=open(oasplFile, 'w'))
+    
+    # initialized avg spectrum
+    avg_response = 0
+    # loop over observers
+    for i in range(numObservers):
+        # get physical time from acoustic data
+        tt = acoustic_data['physical_time']
+        # get acoustic pressures for an observer
+        pp = acoustic_data[i]
+        # convert to dimensional pressure
+        pp = pp * rho * (aSound**2)
+        # get the mean pressure for the observer
+        p0 = np.mean(pp)
+        # get the RMS for the pressure fluctuations
+        p_rms = np.sqrt(np.mean((pp - p0)**2))
+        # get the overall sound pressure based on pressure fluctuations
+        spl = 20 * np.log10(p_rms/p_ref)
+        # print oaspl
+        print('Observer {} | OASPL: {:.5f} dB'.format(i,spl))
+        # print oaspl in the file
+        print("{}, {:.5f}".format(i,spl),file=open(oasplFile, 'a'))
+
+        # get the mean time step size for pressure fluctuations
+        meanStepSize = (tt[1:] - tt[:-1]).mean()
+        # define the windowing function
+        window = np.hanning(len(pp))
+        # windowing checks
+        # window /= window.mean()
+        # window = 1 #no window
+
+        # define the signal
+        h =(pp - p0)*window
+        # fft over the signal
+        X = np.fft.fft(h)
+        # get number of samples
+        N = len(X)
+        n_oneside = N//2
+        # find the dimensional frequencies from the fft
+        freq = np.fft.fftfreq(N,d=meanStepSize/aSound)
+        freq_oneside = freq[:n_oneside]
+        
+        # calculate the sound pressure level
+        response = cal_spl(np.abs(X[:n_oneside]))
+        
+        # collect the spectrum for averaging between observers
+        avg_response += np.abs(response[:n_oneside])
+        
+        # plot the response
+        plot_response(outFileName,freq_oneside,response[:n_oneside],i)
+    
+    # averaging the spectrum across all observers - which is useful in some cases 
+    avg_response = avg_response/numObservers
+    # plot the average spectrum
+    plot_response(outFileName,freq_oneside,avg_response,'avg')
+#end
+
+def main():
+    parser = argparse.ArgumentParser()
+    parser.add_argument('-csv',type=str,required=True)
+    parser.add_argument('-o',type=str,required=True)
```

Comment:
argument types should have two modes: long name and one character name.

the variable name from parser in python code should use long name